Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm

```diff
@@ -398,6 +398,8 @@ static void checkColor(WebCore::CocoaColor *color, std::optional<ColorExpectatio
         EXPECT_WK_STREQ(originalText, contexts.firstObject.attributedText.string);
 
         [[webView writingToolsDelegate] proofreadingSession:session.get() didReceiveSuggestions:@[ adoptNS([[WTTextSuggestion alloc] initWithOriginalRange:NSMakeRange(0, 4) replacement:@"ZZ"]).get() ] processedRange:NSMakeRange(0, 4) inContext:contexts.firstObject finished:NO];
+        TestWebKitAPI::Util::runFor(0.1_s);
```

Comment:
maybe instead of a fixed timeout, there can be a new helper method like `-[WKWebView waitForUpdatedContents]` that can be used in place of this, and just loops until the contents are different. This could also replace your existing `waitForContentValue` method which would also help debugging since errors would show the mismatch instead of just timing out.

Reply:
I don’t understand the second sentence. If I just waiting for content changes, that’s just timing out different if they don’t change.

Reply:
ya if they don't change at all, they'll still time out. But if they change to something that isn't the expected results (which is a relatively frequent type of fail with these types of tests when debugging), they will give an error message with the mismatch instead of timing out.

Reply:
I still don't understand. Are you saying that waitForUpdatedContents should take what we expect it to change _to_, not just wait until the content changes to _anything else_. If so, I can see how that might throw an error. If not, then I don't see any helpful error that it could throw.

Reply:
For example, in this use case

```
        [[webView writingToolsDelegate] compositionSession:session.get() didReceiveText:longerAttributedText.get() replacementRange:NSMakeRange(0, 910) inContext:contexts.firstObject finished:YES];


        [webView waitForContentValue:result];
        EXPECT_WK_STREQ(result, [webView contentsAsStringWithoutNBSP]);
```

if the web view contents do change, but to a value other than `result`, in the current implementation, it will just wait and then time out, and make debugging a bit tricky. In my proposed solution, it will not time out, and instead go on to the next line, which will compare the two values and give an error message that they are different, along with the actual difference between them.

I can make this change in a follow-up PR though so that this can get merged and stop the failing tests.

Reply:
This still won't help in the case where the content never actually changes, Which is what I think it happening in many of the cases. But I could be wrong.